### DISCUSSION TITLE
v1.6.0

### DISCUSSION THREAD

**Discussion by AzulGarza** [Nixtla/neuralforecast#689] [Announcements]
## New Models

- FEDformer

## New features

- Available Mask to specify missing data in input data frame.
- Improve `fit` and `cross_validation` methods with `use_init_models` parameter to restore models to initial parameters.  
- Added robust losses: `HuberLoss`, `TukeyLoss`, `HuberQLoss`, and `HuberMQLoss`.
- Added Bernoulli `DistributionLoss` to build temporal classifiers.
- New `exclude_insample_y` parameter to all models to build models only based on exogenous regressors.
- Added dropout to `NBEATSx` and `NHITS` models. 
- Improved `predict` method of windows-based models to create batches to control memory usage. Can be controlled with the new `inference_windows_batch_size` parameter.
- Improvements to the `HINT` family of hierarchical models: identity reconciliation, `AutoHINT`, and reconciliation methods in hyperparameter selection.
- Added `inference_input_size`hyperparameter to recurrent-based methods to control historic length during inference to better control memory usage and inference times.

## New tutorials and documentation

- Neuralforecast map and How-to add new models
- Transformers for time-series
- Predict insample tutorial
- Interpretable Decomposition
- Outlier Robust Forecasting
- Temporal Classification
- Predictive Maintenance
- Statistical, Machine Learning, and Neural Forecasting methods

## Fixed bugs and new protections
- Fixed bug on `MinMax` scalers that returned NaN values when the mask had 0 values.
- Fixed bug on `y_loc` and `y_scale` being in different devices.
- Added `early_stopping_steps` to the `HINT` method.
- Added protection in the `fit` method of all models to stop training when training or validation loss becomes NaN. Print input and output tensors for debugging.
- Added protection to prevent the case `val_check_step` > `max_steps` from causing an error when early stopping is enabled.
- Added PatchTST to save and load methods dictionaries.
- Added `AutoNBEATSx` to core's `MODEL_DICT`.
- Added protection to the `NBEATSx-i` model where `horizon`=1 causes an error due to collapsing trend and seasonality basis.

<hr /><em>This discussion was created from the release <a href='https://github.com/Nixtla/neuralforecast/releases/tag/v1.6.0'>v1.6.0</a>.</em>
